crates/core/src/mock/mock_starknet.rs

```diff
@@ -201,17 +205,30 @@ pub fn init_testnet_client() -> KakarotClient<SequencerGatewayProvider> {
     let kakarot_address = FieldElement::from_hex_be(KAKAROT_TESTNET_ADDRESS).unwrap();
     let config = StarknetConfig::new(Network::Goerli1Gateway, kakarot_address, Default::default());
 
-    let provider = SequencerGatewayProviderBuilder::new(&Network::Goerli1Gateway).build();
-    KakarotClient::new(config, provider)
+    let provider = Arc::new(SequencerGatewayProviderBuilder::new(&Network::Goerli1Gateway).build());
+    let starknet_account = get_mock_account(provider.clone());
+
+    KakarotClient::new(config, provider, starknet_account)
 }
 
 pub fn init_mock_client(
     fixtures: Option<Vec<StarknetRpcFixture>>,
 ) -> KakarotClient<JsonRpcClient<MockJsonRpcTransport>> {
+    let starknet_provider = Arc::new(mock_starknet_provider(fixtures));
+    let starknet_account = get_mock_account(starknet_provider.clone());
+
     let config = StarknetConfig::new(Network::Katana, *KAKAROT_ADDRESS, *PROXY_ACCOUNT_CLASS_HASH);
-    let starknet_provider = mock_starknet_provider(fixtures);
 
-    KakarotClient::new(config, starknet_provider)
+    KakarotClient::new(config, starknet_provider, starknet_account)
+}
+
+fn get_mock_account<P: Provider + Send + Sync>(provider: P) -> SingleOwnerAccount<P, LocalWallet> {
+    let starknet_account_private_key: FieldElement = SigningKey::from_random().secret_scalar();
+    // Dummy Address
+    let starknet_account_public_address = FieldElement::from(42_u64);
+    let local_wallet = LocalWallet::from_signing_key(SigningKey::from_secret_scalar(starknet_account_private_key));
+
+    SingleOwnerAccount::new(provider, local_wallet, starknet_account_public_address, FieldElement::from(CHAIN_ID))
```

Comment:
Yeah, it makes sense to avoid any confusion, I will make this a local variable inside this function with a comment specifying this is a dummy variable.

I am currently stuck at the rebase of this PR, I will push the changes along-with the rebase.

Reply:
fair!